### PR TITLE
Set up Bintray for released artifact publishing

### PR DESCRIPTION
**Pull request by marshallpierce**
Bintray is the thing behind jcenter (which is the artifact repo you already use). You can choose to sync your files into Maven Central as well for people who are unwilling/unable to use jcenter. (Jcenter is a superset of Maven Central's content, and also has fast CDNs, and a better UI and all that good stuff).

This is a basic setup of Bintray for releasing artifacts. I wasn't sure what artifacts you'd want to include so I just went with the default (jar, source jar, javadoc jar).

I also wasn't sure what naming convention you'd like to use for the final artifact. adoptopenjdk.org isn't actually a domain (it's adoptopenjdk.java.net), so you may wish to change your `group` to that to follow the 'reverse domain' convention.

To use this, you'll need to do the following:
- Sign up for https://bintray.com/. If you want to have this package (jitwatch) live under, say, 'chrisnewland', then you could use the 'maven' repo that they give you by default. If you want it to live organizationally under something like 'adoptopenjdk' then you could create a [Bintray organization](https://bintray.com/docs/usermanual/interacting/interacting_bintrayorganizations.html). This will change the url used to see the Bintray package. For instance, I do not bother with the organization thing for my own packages, so `marshallpierce` is in the URL of this package: https://bintray.com/marshallpierce/maven/org.mpierce.metrics.reservoir%3Ahdrhistogram-metrics-reservoir/view
- Add `bintrayUser` and `bintrayApiKey` to your `~/.gradle/gradle.properties` with values found in the [API key section of your profile](https://bintray.com/profile/edit).
- Set `version` to be a non-snapshot version (e.g. '1.0.0') -- bintray is only for released versions
- Run `./gradlew bintrayUpload`. This should auto-create the 'package' in the Bintray UI for you. You can go and add more details to it, but it should already have the license and URL filled out because that's in the `build.gradle` config. You'll see that it created a '1.0.0' version, and that it's asking you to take a look at the files uploaded and either publish them (so everyone can see) or discard them. If you don't do anything for 6 hours, it will auto-discard.

If you want to set up Maven Central syncing, see http://blog.bintray.com/2014/02/11/bintray-as-pain-free-gateway-to-maven-central/ and https://bintray.com/docs/usermanual/uploads/uploads_syncingartifactswithmavencentral.html.

I also drive-by-fixed a little javadoc thing and the inclusion of .gradle in .gitignore.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -174,12 +176,58 @@ idea {
...
+
+  pkg {
+    repo = 'maven'
+    licenses = ['BSD 2-Clause']
+    vcsUrl = 'https://github.com/AdoptOpenJDK/jitwatch'
+    name = "${project.group}:${project.name}"
+
+    version {
+      name = project.version
+      released = Date.from(java.time.Instant.now())
```
Comment (marshallpierce) [inline build.gradle]: Egregious use of JSR 310 because this is the shortest way I know to get a Date in UTC

Comment (RichardWarburton) [inline build.gradle]: I don't really understand the imports here, but assuming that you're using java.util.Date you can just call new Date(). But a date doesn't have a timezone, so its not in UTC as such, if you call toString() then it'll use the locale in order to get the timezone, if you're relying on that assumption.

Comment (marshallpierce) [inline build.gradle]: Oh, my mistake -- I thought Date had an embedded timezone. I'll just use new Date() then. Thanks!

(re: import -- for some reason I vaguely dislike having import blocks in my build.gradle, especially for classes used once)

**Comment by chriswhocodes** [comment]
Thanks Marshall :) I'll try and get a bintray account for JITWatch set up this week.

**Pull request merged**